.github/workflows/coverage-badges.yml

```diff
@@ -0,0 +1,49 @@
+name: Generate Code Coverage Badges
+
+on:
+  push:
+    branches: [release-* ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Standard build on java 8 with coverage
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'corretto'
+
+      - name: Build with Maven
+        run: mvn -B verify -Dcode-quality --file pom.xml
+
+      - name: Generate JaCoCo badge
+        id: jacoco
+        uses: cicirello/jacoco-badge-generator@v2
+        with:
+          badges-directory: contrib/badges
+          generate-branches-badge: true
+          generate-summary: true
+
+      - name: Log coverage percentages to workflow output
+        run: |
+          echo "coverage = ${{ steps.jacoco.outputs.coverage }}"
+          echo "branches = ${{ steps.jacoco.outputs.branches }}"
+
+      - name: Commit and push the coverage badges and summary file
```

Comment:
FYI this will push a commit to master. If we would rather have a pull request then I can make this generate a PR that we can merge on our own time.